src/content/ui-api/management-api/wc-key-pairs/index.md

```diff
@@ -0,0 +1,264 @@
+---
+
+# Creating workload cluster key pairs via the Management API
+
+We recommend OIDC for authentication to the workload cluster Kubernetes API. However, in some scenarios, X.509 key pairs are a viable alternative. Here we explain a method specific to Giant Swarm to create such key pairs via the Management API.
+
+**Note:** The method described here makes use of the [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) <abbr title="custom resource definition">CRD</abbr></a> and [cert-operator](https://github.com/giantswarm/cert-operator), which are part of every Giant Swarm management cluster currently. As Giant Swarm is shifting towards using Cluster API controllers, this method is not guaranteed to work once this transition is done. Please consider it a temporary solution.
+
+**Note:** If you want to use the deprecated Giant Swarm Rest API for creating key pairs, please check [`gsctl create kubeconfig`]({{< relref "/ui-api/gsctl/create-kubeconfig.md" >}}) and [`gsctl create keypair`]({{< relref "/ui-api/gsctl/create-keypair.md" >}}).
+
+## Prerequisites
+
+You need:
+
+- Your installation's Management API endpoint (similar to `https://g8s.codename.eu-west-1.aws.gigantic.io`)or base domain (similar to `codename.eu-west-1.aws.gigantic.io`).
+- Management API access as an admin.
+- Unique ID of the workload cluster to grant access to.
+- Name of the organization "owning" that workload cluster.
+
+These tools are also required:
+
+- `kubectl`
+- [kubectl gs]({{< relref "/ui-api/kubectl-gs/" >}}), the `kubectl` plug-in for the Giant Swarm Management API.
+
+If you want to use our [example script](#script), you also need:
+
+- [jq](https://stedolan.github.io/jq/)
+- The shell commands/utilities `base64`, `cut`, `date`, and `shasum`
+
+## Ensure Management API access
+
+Using our `kubectl` plug-in, you can make sure to be authenticated against the Management API:
+
+```nohighlight
+kubectl gs login <URL>
+```
+
+Here, `<URL>` can be either
+
+- the Management API endpoint URL or host name
+- your installation's web interface URL or host name
+
+See the [`kubectl gs login` reference]({{< relref " /ui-api/kubectl-gs/login.md" >}}) for details.
+
+## Create a CertConfig resource
+
+The [`CertConfig`]({{< relref "/ui-api/management-api/crd/certconfigs.core.giantswarm.io.md" >}}) resource, which is reconciled by cert-operator, allows to request the creation of a key pair. While it is designed for a number of different purposes, e. g. management cluster components, when configured the right way, the key pair is valid for authentication with a particular workload cluster's Kubernetes API.
+
+Here is an example of a `CertConfig` resource manifest. Follow the detailed explanations below.
+
+```yaml
+apiVersion: core.giantswarm.io/v1alpha1
+kind: CertConfig
+metadata:
+  name: mycert-20210726
+  namespace: default
+  labels:
+    cert-operator.giantswarm.io/version: 0.1.0
```

Comment:
It's pretty brutal customers have to care about this :D

Reply:
Does this need explanation? Do we have to update this every time cert-operator is released? Do we want to remove this and default it via an admission controller?

Reply:
> Does this need explanation?

i have a hunch we can get away with it

> Do we have to update this every time cert-operator is released?

We'd need to update it if we get rid of this cert-operator version, which i don't see happening until we just get rid of cert-operator entirely.

> Do we want to remove this and default it via an admission controller?

Not really :D